openff/toolkit/topology/molecule.py

```diff
@@ -5450,7 +5450,7 @@ def perceive_residues(self, substructure_file_path=None, strict_chirality=True):
             substructure_dictionary_no_chirality = deepcopy(substructure_dictionary)
             # Update inner key (SMARTS) maintaining its value
             for res_name, inner_dict in substructure_dictionary.items():
-                for smarts, atom_types in inner_dict.items():
+                for smarts in inner_dict:
```

Comment:
(not blocking) Slight preference here, this improves readability to my eyes.

```suggestion
                for smarts in inner_dict.keys():
```